src/main/java/com/krishagni/openspecimen/msk2/importer/CarsImporter.java

```diff
@@ -0,0 +1,278 @@
+import com.krishagni.catissueplus.core.biospecimen.services.CollectionProtocolService;
+import com.krishagni.catissueplus.core.common.PlusTransactional;
+import com.krishagni.catissueplus.core.common.errors.OpenSpecimenException;
+import com.krishagni.catissueplus.core.common.events.RequestEvent;
+import com.krishagni.catissueplus.core.common.events.ResponseEvent;
+import com.krishagni.catissueplus.core.common.events.UserSummary;
+import com.krishagni.catissueplus.core.common.service.PermissibleValueService;
+import com.krishagni.catissueplus.core.importer.events.ImportObjectDetail;
+import com.krishagni.catissueplus.core.importer.services.ObjectImporter;
+
+@Configurable
+public class CarsImporter implements ObjectImporter<CarsDetail, CarsDetail> {
+	
+	@Autowired
+	private CollectionProtocolService cpSvc;
+	
+	@Autowired
+	private PermissibleValueService pvSvc;
+	
+	@Autowired
+	private DaoFactory daoFactory;
+	
+	private Map<String, CollectionProtocolDetail> cpMap = new HashMap<>();
+	
+	private Map<String, CollectionProtocolEventDetail> eventsMap = new HashMap<>();
+	
+	private Map<String, SpecimenRequirementDetail> srMap = new HashMap<>();
+	
+	@Override
+	public ResponseEvent<CarsDetail> importObject(RequestEvent<ImportObjectDetail<CarsDetail>> req) {
+		try {
+			ImportObjectDetail<CarsDetail> detail = req.getPayload();
+			importRecord(detail);
+			return ResponseEvent.response(detail.getObject());
+		} catch (OpenSpecimenException ose) {
+			return ResponseEvent.error(ose);
+		} catch (Exception e) {
+			return ResponseEvent.serverError(e);
+		}
+	}
+	
+	@PlusTransactional
+	private ResponseEvent<Object> importRecord(ImportObjectDetail<CarsDetail> detail) throws Exception {
+		CollectionProtocolEventDetail event = toEvent(detail.getObject());
+		
+		CollectionProtocolDetail cpDetail = checkCp(event, detail.getObject());
+		checkEvent(cpDetail, event);
+		checkSr(event);
+		
+		return null;
+	}
+
+	private CollectionProtocolEventDetail toEvent(CarsDetail detail) {
+		CollectionProtocolEventDetail eventDetail = new CollectionProtocolEventDetail();
+		SpecimenRequirementDetail srDetail = new SpecimenRequirementDetail();
+		
+		eventDetail.setCpShortTitle(detail.getIrbNumber());
+		eventDetail.setEventLabel(detail.getCycleName() + detail.getTimepointName());
+		
+		srDetail.setCpShortTitle(detail.getIrbNumber());
+		srDetail.setType(detail.getSpecimenType());
+		srDetail.setStorageType(SPECIMEN_STORAGE_TYPE_VIRTUAL);
+		srDetail.setAnatomicSite(NOT_SPECIFIED);
+		srDetail.setLaterality(NOT_SPECIFIED);
+		srDetail.setPathology(NOT_SPECIFIED);
+		srDetail.setCollectionProcedure(NOT_SPECIFIED);
+		srDetail.setEventLabel(detail.getCycleName() + detail.getTimepointName());
+		srDetail.setName(detail.getProcedureName());
+		srDetail.setInitialQty(new BigDecimal(0));
+		srDetail.setCollectionContainer(detail.getCollectionContainer());
+		srDetail.setSpecimenClass(getSpecimenClass(detail.getSpecimenType()));
```

Comment:
1. The order of fields initialisation will be from higher order to lower order.
2. The higher order fields are those that help in recognising a record from others.
3. In this case, it would be - cp short title, event label, name, type.
4. The rest can appear in whichever order you like